### PR TITLE
DDF-2628 Documentation for the different Solr Catalog Providers

### PR DESCRIPTION
**Pull request by bantillo**
What does this PR do?

Updates the documentation for https://github.com/codice/ddf/pull/1476

Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?

@brjeter 
@oconnormi
@lessarderic 

Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

Solr

Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@figliold
@pklinef

How should this be tested? (List steps with links to updated documentation)

Build documentation.

Any background context you want to provide?

See #1476

What are the relevant tickets?

DDF-2467
DDF-2628

Screenshots (if appropriate)

Checklist:

[x] Documentation Updated
[] Change Log Updated
[] Update / Add Unit Tests
[] Update / Add Integration Tests

### CODE REVIEW

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 1]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+a|
+ 
+|Performance
+a|* Requires no HTTP or network overhead.
+* Near real-time indexing.
+* Can understand complex queries.
+a|
+
+|Backup/Recovery
+a|* Can manually or through custom scripts back up the indexes.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Would this sound better? Backup of indexes can be done manually or via custom scripts.

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 2]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+[cols="1,4a,2a" options="header"]
+|===
+
+|Feature
+|Pro
+|Con
+
+|Scalability
+a|* Allows the middle-tier to be scaled by pointing various middle-tier instances to one server facade.
+* Possible data tier scalability with Solr Cloud. Solr Cloud allows for "high scale, fault tolerant, distributed indexing and search capabilities.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Do we really want to mention Solr Cloud here?  It doesn't seem like it fits in.

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 3]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+|Scalability
+a|* Scalable
+* Can run a cluster of Solr Servers
+a|
+
+|Flexibility
+a|* Can be embedded in Java easily.
+* Uses the same interface as the Standalone Solr Server uses under the covers.
+|* Requires HTTP connection.
+* Setup and Installation is complicateed: needs Zookeeper to coordinate configurations between Solr Servers.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Should this be "complicated"?

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 4]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+* Synchronous issues on startup; i.e., Zookeeper nodes needs to be up and before SolrCloud is up.
+
+|(Administrative)Tools
+a|* Contains Solr Admin GUI, which allows admins to query, check health, see metrics, see configuration files and preferences, etc.
+a|
+
+|Security
+a|
+a|* SolrCloud is not secure, refer to SolrCloud Hardening Guide
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Where is the solr cloud harding guide?  I think a link would be helpful.

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 5]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+ 
+|Performance
+a|* Near real-time indexing
 * Can understand complex queries
-|
+a|* Requires HTTP or network overhead
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Using Requires here sounds strange???

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 1 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: will update

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 2 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: I can remove SolrCloud from here

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 3 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Will fix

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 4 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: @ricklarsen how do I create a link?

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 6]
```diff
@@ -66,30 +48,128 @@ a|* Does not scale. Only runs one single server instance.
...
+
+|Scalability
+a|* Allows the middle-tier to be scaled by pointing various middle-tier instances to one server facade.
+* Possible data tier scalability with Solr Cloud. Solr Cloud allows for "high scale, fault tolerant, distributed indexing and search capabilities.
+a|
+
+|Flexibility
+a|* REST-like HTTP/XML and JSON APIs that make it easy to use from virtually any programming language.
+* Ability to run in separate or same JVM of middle tier.
+a|
```
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: I can add that too, also @lessarderic should I move the Solr Catalog External Provider and Solr Cloud to solr-http-catalog-provider-contents.adoc instead?

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 4 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: if "SolrCloud Hardening Guide" is a section header, you can just add: `<<_solrcloud_hardening_guide,SolrCloud Hardening Guide>>`, but if it's something external, just use the url, followed by the text you want in brackets: `http://example.com[Example Link]`

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 5 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Require's is used all over the documentation, looks like the preferred way to mention use of an internet connection

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 4 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: the hardening guide is not part of DDF documentation

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 7]
```diff
@@ -1,43 +1,25 @@
+=== Solr Catalog Types
+The ${branding} supports 3 Solr Catalog Types:
 
-==== Catalog Solr Embedded Provider
+. Solr Catalog Embedded Provider - included with the ${branding}, useful for testing and demonstration pruposes
+. Solr Catalog External Provider - single Solr Server instance
+. SolrCloud - cluster of standalone Solr Servers
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: `SolrCloud` to `Solr Catalog Cloud Provider` for consistency with the previous two.  This should be applied to all files that use `SolrCloud`.

File: distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc [hunk 8]
```diff
@@ -26,6 +26,52 @@ include::{adoc-include}/_tables/ddf.catalog.solr.external.SolrHttpCatalogProvide
...
+|(Administrative)Tools
+a|* Contains Solr Admin GUI, which allows admins to query, check health, see metrics, see configuration files and preferences, etc.
+* JMX metrics reporting is enabled.
+a|
+
+|Security
+a|* Inherits app server security.
+a|* Web Console must be secured and is openly accessible.
+* REST-like HTTP/XML and JSON APIs must be secured.
+* Current Catalog Provider implementation requires sending unsecured messages to Solr. Without a coded solution, requires network or firewall restrictions in order to secure.
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc]: This is not correct.  Most of these cons are not true if you run it in the DDF container.

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+.SolrCloud Pros and Cons
+[cols="1,4a,2a" options="header"]
+|===
+
+|Feature
+|Pro
+|Con
+
+|Scalability
+a|* Scalable
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Solr Cloud can exceed 2 billion indexed documents.  The other two providers are limited to 2 billion docs but performance will be greatly impacted as you reach that limit.

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+* Can run a cluster of Solr Servers
+a|
+
+|Flexibility
+a|* Can be embedded in Java easily.
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: I do not understand this reason as it relates to the Solr Cloud provider.

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+* Setup and Installation is complicateed: needs Zookeeper to coordinate configurations between Solr Servers.
+* Synchronous issues on startup; i.e., Zookeeper nodes needs to be up and before SolrCloud is up.
+
+|(Administrative)Tools
+a|* Contains Solr Admin GUI, which allows admins to query, check health, see metrics, see configuration files and preferences, etc.
+a|
+
+|Security
+a|
+a|* SolrCloud is not secure, refer to SolrCloud Hardening Guide
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: I would reword this point.  For example "Solr Cloud requires additional network protection to be secure.  See Solr Cloud Hardening Guide for details."

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 7 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: @pklinef does this include the hardening guide?

File: distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc (hunk 8 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc]: will review and remove items that dont make sense

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc (hunk 10 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: will remove

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc (hunk 9 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: will update description

**Comment by pklinef** [comment]
There is the new Solr Catalog Provider that supports embedded, http, and cloud.  Will this PR be updated for that new provider?

**Comment by bantillo** [comment]
@pklinef am not aware of the new Solr Catalog Provider that supports embedded, http, and cloud. This PR only addressed the addition of Solr Cloud and updating the current documentation.

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 12]
```diff
@@ -1,43 +1,25 @@
...
-No configuration is necessary for the embedded Solr Server and the Solr Catalog Provider.
-The standard installation described above is sufficient.
-When the `catalog-solr-embedded-provider` feature is installed, it stores the Solr index files to `<DISTRIBUTION_INSTALLATION_DIRECTORY>/data/solr` by default.
-A user does not have to specify any parameters and the `catalog-solr-embedded-provider` feature contains all files necessary for Solr to start the server. 
-
-However, this component _can_ be configured to specify the directory to use for data storage. 
-
-The configurable properties for the SCP are accessed from the *Catalog Embedded Solr Catalog Provider* configurations in the ${admin-console}.
-
 [TIP]
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: The new Catalog Solr Provider is the default.

https://github.com/codice/ddf/blob/master/catalog/solr/solr-app/src/main/resources/features.xml#L49

But it looks like I forgot to update the metatype to remove the `Catalog External` part before `Solr Catalog Provider`.  That would make it more consistent with the feature and bundle name.

https://github.com/codice/ddf/blob/master/catalog/solr/catalog-solr-provider/src/main/resources/OSGI-INF/metatype/metatype.xml#L18

https://github.com/codice/ddf/blob/master/catalog/solr/catalog-solr-provider/pom.xml#L25

**Comment by pklinef** [comment]
@bantillo See https://github.com/codice/ddf/pull/1478#discussion_r92427133 for details about the new Solr Catalog Provider that supports embedded, http, and cloud.  It is the default provider.

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc (hunk 12 above)
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: I fixed the metatype name for the new Solr Catalog Provider.

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+a|* Scalable
+* Can run a cluster of Solr Servers
+a|
+
+|Flexibility
+a|* Can be embedded in Java easily.
+* Uses the same interface as the Standalone Solr Server uses under the covers.
+|* Requires HTTP connection.
+* Setup and Installation is complicateed: needs Zookeeper to coordinate configurations between Solr Servers.
+* Synchronous issues on startup; i.e., Zookeeper nodes needs to be up and before SolrCloud is up.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: should this say "needs to be up before"?

File: distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc [hunk 14]
```diff
@@ -66,30 +48,30 @@ a|* Does not scale. Only runs one single server instance.
...
-|
+a|* Requires no HTTP or network overhead.
+* Near real-time indexing.
+* Can understand complex queries.
+a|
 
 |Backup/Recovery
-a|* Can manually or through custom scripts back up the indexes
-a|* Must copy files when server is shutdown
+a|* Can manually or through custom scripts back up the indexes.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-catalog-provider-contents.adoc]: Should this say: "Backups of indexes can be preformed manually or through custom scripts"

File: distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc [hunk 15]
```diff
@@ -26,6 +26,52 @@ include::{adoc-include}/_tables/ddf.catalog.solr.external.SolrHttpCatalogProvide
 
 Use the Solr External Provider when the Standalone Solr Server is being used on a separate machine.
 Refer to the Standalone Solr Server recommended configuration.
+Use the Solr Catalog External Provider if using the ${branding} in production mode.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc]: Aren't there two choices for production mode: External or Cloud?

File: distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc [hunk 16]
```diff
@@ -26,6 +26,52 @@ include::{adoc-include}/_tables/ddf.catalog.solr.external.SolrHttpCatalogProvide
...
+The Solr Catalog External Provider is a standalone Solr Server used on a separate machine.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc]: It doesn't have to be on a separate machine...right?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 17]
```diff
@@ -0,0 +1,48 @@
+==== SolrCloud
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+
+If the ${branding} needs to be accessible for a higher than normal period with low instances of downtime, then SolrCloud should be used.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+
+SolrCloud is a cluster of Solr Server instances that support fault tolerance and high availability.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+Each Solr Server instance in SolrCloud provides a Solr Admin GUI.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+
+Synchronization between Solr Server instances is managed by Zookeeper.
+Zookeeper routes client requests to the proper Solr Server instance in the the SolrCloud cluster.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+
+.SolrCloud Pros and Cons
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Should this be Solr Cloud (with space)?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+|Scalability
+a|* Scalable
+* Can run a cluster of Solr Servers
+a|
+
+|Flexibility
+a|* Can be embedded in Java easily.
+* Uses the same interface as the Standalone Solr Server uses under the covers.
+|* Requires HTTP connection.
+* Setup and Installation is complicateed: needs Zookeeper to coordinate configurations between Solr Servers.
```
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: I don't think we should say that the setup is complicated.  Couldn't we just say that it requires the setup of ZooKeeper?

File: distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc (hunk 11 above)
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_sources/solrcloud-contents.adoc]: Make sure to put spaces between the two instances of SolrCloud.

File: distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc (hunk 15 above)
Comment (bantillo) [inline distribution/docs/src/main/resources/_contents/_sources/solr-http-catalog-provider-contents.adoc]: will add

**Comment by bantillo** [comment]
Closing this pull request. Have discused with @lessarderic and the changes will require more changes to the whole Solr documentation. Will also need to incorporate comments from @pklinef into the new PR.
